compiler/plugins/target/AMD-AIE/iree-amd-aie/IR/AMDAIEDmaOpInterface.cpp

```diff
@@ -76,6 +78,23 @@ std::optional<int64_t> getStaticExtent(DoublyStridedOpInterface op) {
   return getAccessRangeExtent(staticSizes.value(), staticStrides.value());
 }
 
+template <CopyOpOperateOn OperateOn>
+std::optional<int64_t> getStaticSize(DoublyStridedOpInterface op) {
+  SmallVector<OpFoldResult> sizes;
+  if constexpr (OperateOn == CopyOpOperateOn::Source) {
+    sizes = op.getSourceMixedSizes();
+  } else if constexpr (OperateOn == CopyOpOperateOn::Target) {
+    sizes = op.getTargetMixedSizes();
+  } else {
+    assert(false && "Function can only operate on Source or Target");
+  }
+  if (sizes.size() == 0) return 0;
+  std::optional<SmallVector<int64_t>> staticSizes = getConstantIntValues(sizes);
+  if (!staticSizes) return std::nullopt;
+  return std::accumulate(staticSizes->begin(), staticSizes->end(), 1,
+                         std::multiplies<>());
```

Comment:
Why do it this way?

We can take the ShapedType from Source/Target and simply `getNumElements()` (can't recall the exact function name but I believe it's this).

Reply:
Aah! Okay. Makes sense. Thanks.